### PR TITLE
Merge isVehicleOnGround and isPedOnGround into one function "isElementOnGround"

### LINKED ISSUE #2650: Merge isVehicleOnGround and isPedOnGround into one function "isElementOnGround"
### Is your feature request related to a problem? Please describe.

not really, just for api quality reason

### Describe the solution you'd like

merge isVehicleOnGround and isPedOnGround into one function "isElementOnGround"

### Describe alternatives you've considered

/

### Additional context

if you have function suitable for more than one type of element, it should have name beginnin with "isElement ""setElement" ect

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not about a cheat or security vulnerability.

### PR DESCRIPTION
**Pull request by nofufubane**
Fixes #2650

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.h [hunk 1]
```diff
@@ -67,6 +67,7 @@ class CLuaElementDefs : public CLuaDefs
     LUA_DECLARE(IsElementDoubleSided);
     LUA_DECLARE(GetElementCollisionsEnabled);
     LUA_DECLARE(IsElementFrozen);
+    bool IsElementOnGround(CClientEntity* entity);
```
Comment (tederis) [inline Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.h]: This method must be static

File: Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp [hunk 2]
```diff
@@ -1558,6 +1559,19 @@ int CLuaElementDefs::IsElementFrozen(lua_State* luaVM)
     return 1;
 }
 
+bool CLuaElementDefs::IsElementOnGround(CClientEntity* entity) {
```
Comment (tederis) [inline Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp:1562]: Although an exception can be thrown, all code paths must return a boolean.

**Review by tederis — COMMENTED** [review]
Looks fine

**Comment by patrikjuvonen** [comment]
Build is failing.

File: Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp (hunk 2 above)
Comment (TracerDS) [reply Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp:1562]: The exception is probably the reason why its not compiling

File: Client/mods/deathmatch/logic/luadefs/CLuaCompatibilityDefs.cpp [hunk 3]
```diff
@@ -23,7 +23,9 @@ void CLuaCompatibilityDefs::LoadFunctions()
         {"isPedInWater", CLuaElementDefs::IsElementInWater},
         {"isPedOnFire", CLuaPedDefs::IsPedOnFire},
         {"setPedOnFire", CLuaPedDefs::SetPedOnFire},
-        {"isPlayerOnGround", CLuaPedDefs::IsPedOnGround},
+        {"isVehicleOnGround", reinterpret_cast<lua_CFunction>(CLuaElementDefs::IsElementOnGround) },
+        {"isPedOnGround", reinterpret_cast<lua_CFunction>(CLuaElementDefs::IsElementOnGround) },
```
Comment (botder) [inline Client/mods/deathmatch/logic/luadefs/CLuaCompatibilityDefs.cpp:27]: It already exists in `CLuaPedDefs::LoadFunctions`

File: Client/mods/deathmatch/logic/luadefs/CLuaCompatibilityDefs.cpp [hunk 4]
```diff
@@ -23,7 +23,9 @@ void CLuaCompatibilityDefs::LoadFunctions()
         {"isPedInWater", CLuaElementDefs::IsElementInWater},
         {"isPedOnFire", CLuaPedDefs::IsPedOnFire},
         {"setPedOnFire", CLuaPedDefs::SetPedOnFire},
-        {"isPlayerOnGround", CLuaPedDefs::IsPedOnGround},
+        {"isVehicleOnGround", reinterpret_cast<lua_CFunction>(CLuaElementDefs::IsElementOnGround) },
```
Comment (botder) [inline Client/mods/deathmatch/logic/luadefs/CLuaCompatibilityDefs.cpp:26]: It already exists in `CLuaVehicleDefs::LoadFunctions`

File: Client/mods/deathmatch/logic/luadefs/CLuaCompatibilityDefs.cpp [hunk 5]
```diff
@@ -23,7 +23,9 @@ void CLuaCompatibilityDefs::LoadFunctions()
...
+        {"isPedOnGround", reinterpret_cast<lua_CFunction>(CLuaElementDefs::IsElementOnGround) },
+        {"isPlayerOnGround", reinterpret_cast<lua_CFunction>(CLuaElementDefs::IsElementOnGround) },
```
Comment (botder) [inline Client/mods/deathmatch/logic/luadefs/CLuaCompatibilityDefs.cpp:28]: You should use the ArgumentParserWarn template here

**Review by botder — CHANGES_REQUESTED** [review]
You should add the deprecated functions to `clientDeprecatedList` in CResourceChecker.Data.h

**Pull request closed without merging**